### PR TITLE
Siamese model code

### PR DESCRIPTION
**Pull request by emwangs**

### CODE REVIEW

**Review by dhan0779 — APPROVED** [review]

**Comment by evanmwilliams** [comment]
Please fix CI check :D

**Comment by evanmwilliams** [comment]
@lisarli Please remove the build files

File: CMakeFiles/3.13.4/CMakeCCompiler.cmake
Comment (evanmwilliams) [inline CMakeFiles/3.13.4/CMakeCCompiler.cmake:1]: @lisarli Please remove the build files

**Review by evanmwilliams — CHANGES_REQUESTED** [review]
Need to clean up repo

**Review by evanmwilliams — APPROVED** [review]
lgtm

**Pull request merged**